chordspy/connection.py

```
@@ -0,0 +1,681 @@
+
+            # Collect data for specified duration
+            while time.time() - start_time < duration:
+                sample, ts = inlet.pull_sample(timeout=1.0)
+                if ts:
+                    timestamps.append(ts)
+
+            if len(timestamps) > 10:
+                diffs = np.diff(timestamps)     # Calculate time differences between consecutive samples
+                filtered_diffs = [d for d in diffs if d > 0]     # Filter out zero/negative differences (invalid)
+                if filtered_diffs:
+                    estimated_rate = 1 / np.mean(filtered_diffs) # Rate = 1/average interval between samples
```

Ruff (0.11.9)

Local variable `estimated_rate` is assigned to but never used

Remove assignment to unused variable `estimated_rate`

(F841)